### PR TITLE
Upgrade sqla dependency specification

### LINKED ISSUE #276: 📦 Update SqlAlchemy package to version >=1.4 < 2.0.0
# Summary
Update SqlAlchemy package to version >1.4.44 < 2.0.0 and address potential issue.

# Details
Updating to the latest SQLA version `1.4.X` where `X` is > `44` does not work out of the box.
Some of the `compliance tests` provided by SQLA are failing. Therefore an update of `sqlalchemy` 
within the context of `sqlalchemy-exasol` also needs address those issues.

## Background & Context
1. Do an overall dependency update
    ```
    poetry update 
    ```
    or just for sqla
    ```
    poetry update SqlAlchemy
    ```

2. Check the test results
    
    ```
    poetry run nox -s integration-tests
    ```

3. Address Issue(s)

    * Fix
    * Disable temporarily (xfail) and create a dedicated issue for investigation/fixing.
    * Suppress/Disable with rational if applicable (e.g. not supported DB feature)
    
## References
* Dependabot PR https://github.com/exasol/sqlalchemy-exasol/pull/274

# Task(s)

- [ ] Update SQLAlchemy to the latest version > 1.4.44 < 2.0.0
- [ ] Address failing sqla compliance tests
- [ ] Bump version number
- [ ] Update change log

### PR DESCRIPTION
**Pull request by Nicoretti**
Fixes #276

- Add pytest-history plugin
- Loosen sqla dependeny constrain
- Adjust changed test name for DifficultParameters tests

### CODE REVIEW

File: test/integration/regression/test_regression_bug335.py [hunk 1]
```diff
@@ -65,17 +68,21 @@ def test_lastrowid_does_not_create_extra_commit(
         schema=schema_name,
     )
 
-    with odbcconfig(ODBC_DRIVER):
-        conn = engine.connect()
-        trans = conn.begin()
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=sqlalchemy.exc.RemovedIn20Warning)
```
Comment (kaklakariada) [inline test/integration/regression/test_regression_bug335.py:72]: Which warnings does this filter? I am just curious

File: test/integration/sqlalchemy/test_suite.py [hunk 2]
```diff
@@ -154,25 +176,33 @@ def test_update_rowcount_return_defaults(self, connection):
...
         ("1col:on",),
-        argnames="name",
+        argnames="paramname",
     )
-    def test_round_trip(self, name, connection, metadata):
-        super().test_round_trip(name, connection, metadata)
+
+    @tough_parameters
+    def test_round_trip_same_named_column(self, paramname, connection, metadata):
+        # dot_s and qmarks are currently disabled see https://github.com/exasol/sqlalchemy-exasol/issues/232
```
Comment (kaklakariada) [inline test/integration/sqlalchemy/test_suite.py:204]: You removed `q?marks`. `dot_s` was never enabled. Maybe add a note note #232 to add & fix these tests. Currently the ticket only says to enable the test.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: test/integration/sqlalchemy/test_suite.py [hunk 3]
```diff
@@ -154,25 +176,33 @@ def test_update_rowcount_return_defaults(self, connection):
...
         ("more :: %colons%",),
+        ("_name",),
+        ("___name",),
+        ("[BracketsAndCase]",),
+        ("42numbers",),
+        ("percent%signs",),
+        ("has spaces",),
         ("/slashes/",),
         ("more/slashes",),
-        ("q?marks",),
```
Comment (tkilias) [inline test/integration/sqlalchemy/test_suite.py:169]: why did you remove this?

File: test/integration/sqlalchemy/test_suite.py [hunk 4]
```diff
@@ -154,25 +176,33 @@ def test_update_rowcount_return_defaults(self, connection):
...
         ("1param",),
         ("1col:on",),
-        argnames="name",
+        argnames="paramname",
```
Comment (tkilias) [inline test/integration/sqlalchemy/test_suite.py:199]: why did tis change?

**Review by tkilias — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

File: test/integration/sqlalchemy/test_suite.py (hunk 4 above)
Comment (Nicoretti) [reply test/integration/sqlalchemy/test_suite.py:199]: Was changed by SQLA

**Review by tkilias — APPROVED** [review]
